crates/orderbook/src/database/orders.rs

```diff
@@ -76,13 +76,44 @@ pub trait OrderStoring: Send + Sync {
         limit: Option<u64>,
     ) -> Result<Vec<Order>>;
     async fn latest_order_event(&self, order_uid: &OrderUid) -> Result<Option<OrderEvent>>;
+    async fn single_order_with_quote(&self, uid: &OrderUid) -> Result<Option<OrderWithQuote>>;
 }
 
 pub struct SolvableOrders {
     pub orders: Vec<Order>,
     pub latest_settlement_block: u64,
 }
 
+pub struct OrderWithQuote {
+    pub order: Order,
+    pub quote: Option<orders::Quote>,
+}
+
+impl OrderWithQuote {
+    pub fn new(order: Order, quote: Option<Quote>) -> Self {
+        Self {
+            quote: quote.map(|quote| orders::Quote {
+                order_uid: ByteArray(order.metadata.uid.0),
+                gas_amount: quote.data.fee_parameters.gas_amount,
+                gas_price: quote.data.fee_parameters.gas_price,
+                sell_token_price: quote.data.fee_parameters.sell_token_price,
+                sell_amount: u256_to_big_decimal(&quote.sell_amount),
+                buy_amount: u256_to_big_decimal(&quote.buy_amount),
+                solver: ByteArray(quote.data.solver.0),
+            }),
+            order,
+        }
+    }
+}
+
+impl Deref for OrderWithQuote {
```

Comment:
Let's not try to get fancy here. `Deref` is mostly "reserved" for smart pointer types and things like drop guards.

Reply:
1. Initially the docs [specifically mentioned](https://github.com/rust-lang/rust/issues/91004) to only use if for smart pointers.
2. since then the phrasing was relaxed to `users of that type should not get surprised by deref` which arguable is not the case here because we almost never use `Deref` (mostly a point for code consistency)

> now it is super ugly with the order.order
Sure, but it also doesn't look like you tried hard to avoid it looking ugly...
```rust
let OrderWithQuote{order, quote} = weird_new_struct;
```

IMO requiring this "hack" rather indicates that the chosen abstraction doesn't work great.
For example you could also consider making the quote part of the `FullOrder` which probably is useful in more than just this specific case.